cdap-proto/src/main/java/co/cask/cdap/proto/security/Principal.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.security;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.proto.id.EntityId;
+
+/**
+ * Identifies a user, group or role which can be authorized to perform an {@link Action} on a {@link EntityId}.
+ */
+@Beta
+public class Principal {
+  /**
+   * Identifies the type of {@link Principal}.
+   */
+  public enum PrincipalType {
+    USER,
+    GROUP,
+    ROLE
```

Comment:
We discussed offline that having the same object represent both ``USER/GROUP`` and ``ROLE`` may create problems in future if we have APIs of the form ``addUserToRole(Principal user, Principal role)``.  We should move ``Role`` into a separate class, so such APIs can become less ambiguous as ``addUserToRole(Principal user, Role role)``. I will make this change in a later PR though to think it through more thoroughly, since it would mean a overloading a lot of APIs in ``Authorizer``.